### PR TITLE
HW07 Manninen Igor

### PR DESCRIPTION
**Pull request by imanninen**
Spent around 11-12 hours on this homework.

### CODE REVIEW

File: hw07/src/Parser.hs [hunk 1]
```diff
@@ -156,10 +124,75 @@ parsePair' = inParens '(' ')'
...
+  fmap Const parseInt
+
+parseVar :: Parser (Expr Int)
+parseVar = do
+  fmap Var parseIdent
+
+satisfySqrtKeyWord :: Parser [Char]
+satisfySqrtKeyWord = Parser $ \str ->
+  case str of
+    s:q:r:t:l | [s, q, r, t] == "sqrt" -> Right ( l, "sqrt")
```
Comment (kajigor) [inline hw07/src/Parser.hs:185]: You can pattern match on chars directly: 

```
case str of 
  's' : 'q' : 'r' : 't' : l -> Right (l, "sqrt") 
```

**Review by kajigor — CHANGES_REQUESTED** [review]

File: hw07/src/Parser.hs [hunk 2]
```diff
@@ -2,151 +2,119 @@
...
-    t <- go              -- then follows a sequence of letters or digits
-    return (h : t)       -- we compose the result 
-  where 
+parseIdent :: Parser String
+parseIdent = do
+    h <- satisfy isAlpha "is not an alpha"
+    t <- go
+    let name = h:t       
+    if name `elem` keywords 
+      then failParser "sqrt is not an ident!!" -- there it should fails but it not :(
```
[the remark this answers is not in the document]
Comment (imanninen) [reply hw07/src/Parser.hs:77]: sqrt is not an Identifier, and that is what it said. Or I miss something? Or if you about this comment, I forgot to remove it :)

File: hw07/app/Main.hs [hunk 3]
```diff
@@ -1,4 +1,24 @@
 module Main (main) where
+import Expr (Expr(..), BinOpeation(..), runEval, UnoOperation(..))
+import Parser (Parser(..), runParser, parseExpression, parseBinOp)
+
+myApp :: IO ()
+myApp = do
+  putStrLn "Please enter the expression in presix notation to parse it!"
+  line <- getLine
+  case line of 
+    "exit" -> putStrLn "Bye!"
```
[the remark this answers is not in the document]
Comment (imanninen) [reply hw07/app/Main.hs:10]: sad(

File: hw07/src/Parser.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kajigor) [reply hw07/src/Parser.hs]: > if you about this comment, I forgot to remove it :)

Yep, it behaved as expected contrary to the comment

**Comment by kajigor** [comment]
ok

**Pull request closed without merging**